Wall+Werror

Would it be possible to add `-Werror` to the existing `-Wall` in the developer CI workflow? I don't see very many warnings in the output, so if we can resolve those, that would let us enable `-Werror` to protect against new code issues, and help address https://github.com/NOAA-EMC/NCEPLIBS/issues/191.

I tried compiling wgrib2 with -Werrors and found a problem.

/cpc/home/wd51we/wgrib2_ens_qc/wgrib2/wgrib2/New_grid.c:50:20: error:
‘UV_vectors’ defined but not used [-Werror=unused-variable]
 static const char *UV_vectors[] = { "UGRD", "VGRD", NULL };

The code New_grid.c there is a reference:

            save->uv_vectors = (const char **) UV_vectors;

So the compiler is giving a fake warning.

I was trying to remove a warning from the fv3 model and the
compiler was complaining that a variable wasn't initialized.
However, the variable was initialized through a pointer.

Another problem in the fv3 was the model uses a parent_grid for
regional models.  So a regional model has a parent grid.
A global model has an undefined parent_grid.  The
compiler was complaining that the parent_grid was undefined.









On Wed, Sep 10, 2025 at 6:50 PM Alex Richert ***@***.***>
wrote:

> *AlexanderRichert-NOAA* created an issue (NOAA-EMC/wgrib2#444)
> <https://github.com/NOAA-EMC/wgrib2/issues/444>
>
> Would it be possible to add -Werror to the existing -Wall in the
> developer CI workflow? I don't see very many warnings in the output, so if
> we can resolve those, that would let us enable -Werror to protect against
> new code issues, and help address NOAA-EMC/NCEPLIBS#191
> <https://github.com/NOAA-EMC/NCEPLIBS/issues/191>.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/wgrib2/issues/444>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEIB7ZXMMP6JB5PQZNO5UW33SBXITAVCNFSM6AAAAACGFGUEDCVHI2DSMVQWIX3LMV43ASLTON2WKOZTGQYDGNJWG4YTSOA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

--- a/.github/workflows/developer.yml
+++ b/.github/workflows/developer.yml
@@ -60,9 +60,9 @@ jobs:
         cd wgrib2
         mkdir build
         cd build
-        export CFLAGS='-Wall -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -I/home/runner/work/wgrib2/wgrib2/nceplibs/NCEPLIBS-g2c/include'
-        export FCFLAGS='-Wall -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0'
-        export FFLAGS='-Wall -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0'
+        export CFLAGS='-Wall -Werror -Wno-error=unused-parameter -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -I/home/runner/work/wgrib2/wgrib2/nceplibs/NCEPLIBS-g2c/include'
+        export FCFLAGS='-Wall -Werror -Wno-error=unused-parameter -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0'
+        export FFLAGS='-Wall -Werror -Wno-error=unused-parameter -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0'
         cmake .. -DENABLE_DOCS=ON -DFTP_TEST_FILES=ON -DCMAKE_PREFIX_PATH="$GITHUB_WORKSPACE/nceplibs/NCEPLIBS-ip;$GITHUB_WORKSPACE/nceplibs/jasper;$GITHUB_WORKSPACE/nceplibs/NCEPLIBS-g2c" -DTEST_FILE_DIR=/home/runner/data -DMAKE_FTN_API=ON -DUSE_NETCDF=ON -DUSE_AEC=ON -DUSE_IPOLATES=ON -DUSE_G2CLIB_LOW=ON -DUSE_OPENMP=ON -DBUILD_WGRIB=OFF
         make VERBOSE=1
         ctest --verbose --output-on-failure --rerun-failed

--- a/.github/workflows/developer.yml
+++ b/.github/workflows/developer.yml
@@ -60,9 +60,9 @@ jobs:
         cd wgrib2
         mkdir build
         cd build
-        export CFLAGS='-Wall -Werror -Wno-error=unused-parameter -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -I/home/runner/work/wgrib2/wgrib2/nceplibs/NCEPLIBS-g2c/include'
-        export FCFLAGS='-Wall -Werror -Wno-error=unused-parameter -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0'
-        export FFLAGS='-Wall -Werror -Wno-error=unused-parameter -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0'
+        export CFLAGS='-Wall -Werror -Wno-error=unused-parameter -Wno-error=unused-dummy-argument -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -I/home/runner/work/wgrib2/wgrib2/nceplibs/NCEPLIBS-g2c/include'
+        export FCFLAGS='-Wall -Werror -Wno-error=unused-parameter -Wno-error=unused-dummy-argument -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0'
+        export FFLAGS='-Wall -Werror -Wno-error=unused-parameter -Wno-error=unused-dummy-argument -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0'
         cmake .. -DENABLE_DOCS=ON -DFTP_TEST_FILES=ON -DCMAKE_PREFIX_PATH="$GITHUB_WORKSPACE/nceplibs/NCEPLIBS-ip;$GITHUB_WORKSPACE/nceplibs/jasper;$GITHUB_WORKSPACE/nceplibs/NCEPLIBS-g2c" -DTEST_FILE_DIR=/home/runner/data -DMAKE_FTN_API=ON -DUSE_NETCDF=ON -DUSE_AEC=ON -DUSE_IPOLATES=ON -DUSE_G2CLIB_LOW=ON -DUSE_OPENMP=ON -DBUILD_WGRIB=OFF
         make VERBOSE=1
         ctest --verbose --output-on-failure --rerun-failed

